impeller/toolkit/android/proc_table.h

```diff
@@ -0,0 +1,125 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_TOOLKIT_ANDROID_PROC_TABLE_H_
+#define FLUTTER_IMPELLER_TOOLKIT_ANDROID_PROC_TABLE_H_
+
+#include <android/api-level.h>
+#include <android/hardware_buffer.h>
+#include <android/hardware_buffer_jni.h>
+#include <android/surface_control.h>
+#include <android/trace.h>
+
+#include <functional>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/native_library.h"
+
+namespace impeller::android {
+
+#define FOR_EACH_ANDROID_PROC(INVOKE)            \
+  INVOKE(ATrace_isEnabled, 23)                   \
+  INVOKE(AChoreographer_getInstance, 24)         \
+  INVOKE(AChoreographer_postFrameCallback, 24)   \
```

Comment:
We must not use this if `sizeof(uint64_t) != sizeof(long)`.

On 32 bit platforms we have to use the Java based API. But we still should try to resolve it on a 64 bit platform that is < API 29.  I made the same mistake when migrating things to the NDKHelpers class :)

Reply:
(That's why the `#if FML_ARCH_64BIT` guard was around this, or whatever it actually was)